src/NuGet.Core/NuGet.Build.Tasks.Console/DependencyGraphSpecGenerator.cs

```diff
@@ -0,0 +1,802 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+
+#if IS_CORECLR
+using System.Runtime.InteropServices;
+#endif
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Definition;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Evaluation.Context;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Graph;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Credentials;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging;
+using NuGet.ProjectModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.RuntimeModel;
+using NuGet.Versioning;
+using ILogger = Microsoft.Build.Framework.ILogger;
+
+namespace NuGet.Build.Tasks.Console
+{
+    internal class DependencyGraphSpecGenerator
+    {
```

Comment:
I did not see tests for the new types.

Reply:
I will be adding unit tests in a future PR